Ictt course (#37)
ava-labs/avalanche-academy: content/course/interchain-token-transfer/05-avalanche-interchain-token-transfer/02-bridge-design.mdx

---
title: Interchain Token Transfer Design
description: Get familiar with the ICTT design
updated: 2024-05-31
authors: [ashucoder9]
icon: BookOpen
---

> Each token transferrer instance consists of one "home" contract and at least one but possibly many "remote" contracts. Each home contract instance manages one asset to be transferred out to `TokenRemote` instances. The home contract lives on the Subnet where the asset to be transferred exists. A transfer consists of locking the asset as collateral on the home Subnet and minting a representation of the asset on the remote Subnet. The remote contracts, each of which has a single specified home contract, live on other Subnets that want to import the asset transferred by their specified home. The token transferrers are designed to be permissionless: anyone can register compatible `TokenRemote` instances to allow for transferring tokens from the `TokenHome` instance to that new `TokenRemote` instance. The home contract keeps track of token balances transferred to each `TokenRemote` instance, and handles returning the original tokens back to the user when assets are transferred back to the `TokenHome` instance. `TokenRemote` instances are registered with their home contract via a Teleporter message upon creation.

Reviewer: ```suggestion
Each token transferrer instance consists of one "home" contract and at least one but possibly many "remote" contracts. Each home contract instance manages one asset to be transferred out to `TokenRemote` instances. The home contract lives on the Subnet where the asset to be transferred exists.

A transfer involves locking the asset as collateral on the home Subnet and minting a representation of the asset on the remote Subnet. The remote contracts, each with a single specified home contract, live on other Subnets that want to import the asset transferred by their specified home.

The token transferrers are designed to be permissionless: anyone can register compatible `TokenRemote` instances to transfer tokens from the `TokenHome` instance to that new `TokenRemote` instance.

The home contract keeps track of token balances transferred to each `TokenRemote` instance and handles returning the original tokens to the user when assets are transferred back to the `TokenHome` instance.

`TokenRemote` instances are registered with their home contract via a Teleporter message upon creation.
```
